src/data/markdown/docs/05 Examples/01 Examples/24 distribute-workloads.md

```diff
@@ -0,0 +1,229 @@
+
+To account for these different behaviors, testers often want to distribute their VUs into different logical behaviors.
+k6 supports multiple ways to do this.
+The following sections provide different ways to split logic across VUs, along with some strategies to granularly filter results.
+
+<Blockquote mod="note" title="Aim for simplicity">
+
+These techniques could be used to create very complex configurations.
+However, more complexity creates more ambiguity for results interpretation
+
+</Blockquote>
+
+
+## Split logic across scenarios
+
+<Blockquote mod="note" title="">
+
+In this context, _workload_ refers to the traffic pattern simulated by a scenario.
+
+</Blockquote>
+
+
+One way to distribute traffic is to use scenarios to schedule different workloads for different functions.
+1. Define multiple scenarios in your `options.scenarios` object.
+1. Use the scenario `exec` property to execute different VU functions with a workload.
+
+For example, imagine a social media site that typically receives 100 concurrent users.
+Of those, 80 might visit their contacts page, and 20 might view the news.
+To configure such a distribution, make two scenarios with different throughput or VUs:
+
+```javascript
+import http from "k6/http";
+
+export const options = {
+  //scenario to view contacts
+  scenarios: {
+    contacts: {
+      executor: "shared-iterations",
+      exec: "contacts",
+      vus: 80,
+      iterations: 100,
+    },
+    //scenario to view news
+    news: {
+      executor: "shared-iterations",
+      exec: "news",
+      vus: 20,
+      iterations: 100,
+    },
+  },
+};
+
+//use the exec property to run different scenarios for different functions
+
+export function contacts() {
+  http.get("https://test.k6.io/contacts.php");
+}
+
+export function news() {
+  http.get("https://test.k6.io/news.php");
+}
+```
+
+### Tip: reduce scenario boilerplate with `new`
+
+If your script uses multiple scenarios, consider making a function to construct a scenario object.
+
+For example, the following script makes the same test as the preceding section.
+But instead of a new scenario each time, it uses the `new` operator to construct new workloads based on arguments for the number of VUs and `exec` name.
+
+<Collapsible title="workload constructor" isOpen="" tag="">
+
+<CodeGroup labels={["workload-constructor.js"]} lineNumbers={[true]} showCopyButton={[true]}>
+
+
+```javascript
+import http from "k6/http";
+
+// workload constructor
+function Scenario(vus, exec) {
+  this.executor = "shared-iterations";
+  this.vus = vus;
+  this.exec = exec;
+  this.iterations = 100;
+}
+
+//create scenario object
+const contactWorkload = new Scenario(80, "contacts");
+const newsWorkload = new Scenario(20, "news");
+
+export const options = {
+  //scenario to view contacts
+  scenarios: {
+    contactWorkload,
+    newsWorkload,
```

Comment:
nit: I would guess most people will find this syntax confusing, so using `"contact": contactWorkload` is likely better idea. 

Also you don't really need a constructor and using `new` to be honest - you can just have a function (which you do).

Reply:
> you don't really need a constructor and using new to be honest - you can just have a function (which you do)

I'm going to remove this section, as I commented https://github.com/grafana/k6-docs/pull/1179/files#r1196400374 ,

But as an aside, my understanding is that `new` is never necessary, but it helps make behavior more explicit. That's how I understand this page, and I've seen `new` used in a few modern JS libraries so I assumed it was considered good practice.